analysis/import/SourceCodeParser/src/main/kotlin/de/maibornwolff/codecharta/importer/sourcecodeparser/SourceCodeParserMain.kt

```diff
@@ -116,6 +119,15 @@ class SourceCodeParserMain(
         val pipedProject = ProjectDeserializer.deserializeProject(input)
         writer.generate(projectParser.projectMetrics, projectParser.metricKinds, pipedProject)
 
+        outputFile?.let { nonNullOutputFile ->
+            val absoluteFilePath = if (writer::class == CSVMetricWriter::class) {
+                OutputFileHandler.checkAndFixFileExtensionCsv(nonNullOutputFile.absolutePath)
+            } else {
+                OutputFileHandler.checkAndFixFileExtensionJson(nonNullOutputFile.absolutePath, compress)
+            }
+            logger.info("Created output file at $absoluteFilePath")
+        }
```

Comment:
Please check if this new code can be moved into a new method to not extend the call method any further.